libs/nio/src/main/java/org/elasticsearch/nio/Page.java

```diff
@@ -61,16 +61,21 @@ public void close() {
 
     private static class RefCountedCloseable extends AbstractRefCounted {
 
-        private final Runnable closeable;
+        private final Closeable closeable;
 
-        private RefCountedCloseable(Runnable closeable) {
+        private RefCountedCloseable(Closeable closeable) {
             super("byte array page");
             this.closeable = closeable;
         }
 
         @Override
         protected void closeInternal() {
-            closeable.run();
+            try {
```

Comment:
Maybe this pattern should be another util in `Releasables`?

Reply:
I thought about this yesterday but decided not to. We have one spot where we use this with `Closeable` still (`InboundMessage`) and can't change that either and  Maybe it would make more sense to do a follow-up where we add this pattern to `closeInternal` globally (that really shouldn't ever throw IMO but currently does in a number of spots) which would cover almost all the duplication here anyway?

Reply:
I think I'd rather do it now, IMO it doesn't really matter that we can't fix that one `Closeable` case yet, and I think it's a useful pattern for other uses of Releasable too.

Reply:
Alrighty, added it here turned out much nicer than expected :) (just one closable case and `Page` which doesn't have access to `Releasables` that this didn't apply to).

Reply:
IMO it makes sense to move `Releasables` next to `Releasable` and then we could use it here too. Unless `Releasables` has some other trickier dependences that I can't see (I didn't try it)

Reply:
++ moved cleanly :) I also added the single item optimization we have for the normal `close` to it now that this is used in a number of spots. See  1625466